File: backend/src/main/java/com/cdac/farmermarketplace/service/ReviewService.java
package com.cdac.farmermarketplace.service;


import java.util.List;

import com.cdac.farmermarketplace.dto.request.ReviewRequest;
import com.cdac.farmermarketplace.dto.response.ReviewResponse;



public interface ReviewService {


    // Add review for product
    ReviewResponse addReview(
            Long productId,
            ReviewRequest request,
            Long userId
    );



    // Get all reviews of product
    List<ReviewResponse> getProductReviews(
            Long productId
    );



    // Delete own review
    void deleteReview(
            Long reviewId,
            Long userId
    );



    // Update own review
    ReviewResponse updateReview(
            Long reviewId,
            ReviewRequest request,
            Long userId
    );
    
    Double getAverageRating(Long productId);


    Long getReviewCount(Long productId);


}

File: frontend/src/services/reviewService.js
import api from "./api";


// =====================================================
// GET PRODUCT REVIEWS
// =====================================================

export const getProductReviews = async (productId) => {

    const response = await api.get(
        `/reviews/product/${productId}`
    );

    return response.data;
};



// =====================================================
// ADD REVIEW
// =====================================================

export const addReview = async (
    productId,
    reviewData
) => {

    const response = await api.post(
        `/reviews/product/${productId}`,
        reviewData
    );

    return response.data;
};



// =====================================================
// UPDATE REVIEW
// =====================================================

export const updateReview = async (
    reviewId,
    reviewData
) => {

    const response = await api.put(
        `/reviews/${reviewId}`,
        reviewData
    );

    return response.data;
};



// =====================================================
// DELETE REVIEW
// =====================================================

export const deleteReview = async (
    reviewId
) => {

    const response = await api.delete(
        `/reviews/${reviewId}`
    );

    return response.data;
};